gedi/process_l2a.py

```diff
@@ -227,57 +273,155 @@ def df_to_geojson(df, outfile):
-    ]
-
-    df = gedi_L2A_to_df(args.dir, files, bbox)
-    append_canopy_metrics(df, canopy_threshold=2)
-    del df["rh"]
-    if args.filetype.lower() == "csv":
-        df.to_csv(f"{args.dir}{args.outfile}.csv", index=False)
-    elif args.filetype.lower() == "parquet":
-        df.to_parquet(f"{args.dir}{args.outfile}.parquet.gzip", compression="gzip")
-    elif args.filetype.lower() == "geojson":
-        df_to_geojson(df, f"{args.dir}{args.outfile}.geojson")
+    print("------------------------------------------")
+
+    bbox = [float(b) for b in args.bbox.split(',')]
+
+    df = pd.DataFrame()
+
+    with open(str(args.textfile), 'r') as texturls:
+        for url in texturls:
+            if validators.url(url):
+
+                status = download_url(url.rstrip('\n'), args.dir)
+                if not status:
+                    print("------------------------------------------")
+                    continue
+                print("          ----------------------          ")
+                # Make list of filepaths
+                filepaths = [
+                    os.path.join(root, file)
+                    for root, dir, files in os.walk(args.dir, topdown = False)
+                    for file in files
+                    if file.endswith('.h5')
+                ]
+
+                tmp_df = gedi_L2A_to_df(filepaths, bbox = bbox)
+                print("          ----------------------          ")
+                # append data to dataframe
+                df = df.append(tmp_df)
+                print(f"Appended {len(tmp_df)} lines to dataframe; Dataframe has {len(df)} lines")
+
+                # Delete hdf5 files
+                for filepath in filepaths:
+                    rmtree(os.path.dirname(filepath))
+
+                print("------------------------------------------")
+
+    if df.empty:
+        print("DataFrame is empty. Not writing data to file")
+        print("------------------------------------------")
     else:
-        raise ValueError(
-            f"Received unsupported file type {args.filetype}. Please provide one "
-            "of: csv, parquet, or GeoJSON."
-        )
+        if args.filetype.lower() == "csv":
+            filename = os.path.join(args.dir, args.outfile + ".csv")
+            print(f'Writing to file {filename}')
+            df.to_csv(filename, index=False)
+        elif args.filetype.lower() == "parquet":
+            filename = os.path.join(args.dir, args.outfile + ".parquet.gzip")
+            print(f'Writing to file {filename}')
+            df.to_parquet(filename, compression="gzip")
+        elif args.filetype.lower() == "geojson":
+            filename = os.path.join(args.dir, args.outfile + ".geojson")
+            print(f'Writing to file {filename}')
+            df_to_geojson(df, filename)
+        else:
+            raise ValueError(
+                f"Received unsupported file type {args.filetype}. Please provide one of: csv, parquet, or GeoJSON."
+            )
```

Comment:
It might make more sense to do the file extension handling before the downloading and unpacking, just in case a user provides the wrong file extension (or has a typo), to save the risk of losing that downloading and unpacking. Another option would be to store the output of the intermediate steps, but I think just checking up-front probably makes more sense than that to me.